code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm

```diff
@@ -213,3 +228,237 @@
 		to_chat(user, "<span class='warning'>There is no active AI within \the [ai].</span>")
 	update_verb_holder()
 	return
+
+/*
+ * The realm of cyberpunk, a Neural Jack-enabled IIS.
+ * Components unashamedly utilizing the groundwork of borers.
+ */
+/obj/item/ai_verbs/advanced/verb/assume_control()
+	set category = "Hardsuit"
+	set name = "Enable Direct Control"
+	set src in usr
+
+	if(!usr.loc || !usr.loc.loc || !istype(usr.loc.loc, /obj/item/rig_module))
+		to_chat(usr, "You are not loaded into a hardsuit.")
+		return
+
+	var/obj/item/rig_module/ai_container/advanced/module = usr.loc.loc
+	if(!module.holder)
+		to_chat(usr, "Your module is not installed in a hardsuit.")
+		return
+
+	module.jack_brain()
+
+/obj/item/rig_module/ai_container/advanced
+	name = "AINI module"
+	desc = "An artificial intelligence neural integration module, suitable for most hardsuits."
+	description_fluff = "The unassuming device has no exterior markings, save for a network usage monitor and tiny warning label. On closer inspection, it reads, \"Warning: Device may cause damage if used without proper neural interface firmware.\" Something in your core urges you to heed that warning."
+	description_info = "Unless used by a robotic RIG pilot, or an organic one with a neural implant, the module will cause brain-damage on activation of its override mode."
+	icon_state = "AIIS"
+	var/controlling = FALSE
+
+	verb_holder = /obj/item/ai_verbs/advanced
+
+	var/last_defib = 1
+	var/mob/living/captive_brain/pilot_brain	
+	var/obj/item/weapon/shockpaddles/rig/paddles
+	var/imperfect_connection = FALSE	// Is the AI properly connected to the body? If N, they can't speak without sign-lang.
+	var/corpse_pilot = FALSE	// Is the AI controlling a corpse? If Y, kill the body again when they revert.
+
+/obj/item/rig_module/ai_container/advanced/Initialize()
+	. = ..()
+
+	paddles = new(src)
+
+/obj/item/rig_module/ai_container/advanced/removed()
+	revert()
+	..()
+
+/obj/item/rig_module/ai_container/advanced/process()
+	if((holder.offline || !holder.ai_override_enabled) && controlling)
+		revert()
+
+	if(controlling)
+		var/mob/living/carbon/human/H = holder.wearer
+
+		var/mob/living/simple_mob/animal/borer/Borer = H.has_brain_worms()
+		if(Borer)	// As funny as it would be to see a borer, AI, and human swap minds like musical chairs, let's not.
+			revert()
+			to_chat(integrated_ai, "<span class='danger'>The neural jack signals a warning abruptly, before rapidly retracting.</span>")
+			to_chat(H, "<span class='warning'>You are shoved into consciousness as though you have been dropped into freezing water.</span>")
+			return
+
+		var/obj/item/organ/internal/brain/Brain = H.internal_organs_by_name[O_BRAIN]
+		if(Brain)
+			if(Brain.robotic < ORGAN_ASSISTED)
+				H.add_modifier(/datum/modifier/mute, 30 SECONDS, suppress_failure = TRUE)
+
+		if(H.stat == DEAD && world.time > last_defib + 1 MINUTE && (H.getBruteLoss() + H.getFireLoss()) < (H.getMaxHealth() * 2 - H.getCloneLoss() * 2))
```

Comment:
This is a lot of boilerplate. Can't you call out to the defib to apply all this somehow rather than reimplementing?

Reply:
This is the least-boiler of the methods I found. It needed to circumnavigate the normal checks the DEFIB does, and this method allows the proper associated power-draw.

Reply:
What specific checks by the defib need to be avoided? Can you refactor these checks to a separate proc on the defib?

Reply:
The standard attackby chain.

Reply:
Math oofs and, maxhealth specifically should be doubled as humanoids go into the negative for their health calculation.

Reply:
Please anwer the previous question: What checks in the standard attackby chain are you specifically trying to avoid here?